### PR TITLE
[Orc] Add NotifyCreated callback for LLJITBuilder

### PR DESCRIPTION
**Pull request by weliveindetail**
This is useful to attach generators to JITDylibs or inject initial symbol definitions.

### CODE REVIEW

**Comment by weliveindetail** [comment]
It seems like there are no tests for `LLJITBuilder` so far.

**Comment by weliveindetail** [comment]
Please note the differences to the existing callbacks `SetupProcessSymbolsJITDylib` and `SetUpPlatform`. They receive a reference to the `LLJIT` instance as well, but they both:
* are called during construction, i.e. before the last point of failure
* have a well-defined purpose that we would have to mimic
* have a default definition, that is inaccessible via the interface (e.g. `setUpGenericLLVMIRPlatform`)

**Review by lhames — APPROVED** [review]
I think it makes sense for `NotifyCreated` to return `Error` rather than `void` so that the callback itself can fail. Otherwise this looks great -- thanks @weliveindetail!

File: llvm/include/llvm/ExecutionEngine/Orc/LLJIT.h [hunk 1]
```diff
@@ -474,6 +487,11 @@ class LLJITBuilderSetters {
     std::unique_ptr<JITType> J(new JITType(impl(), Err));
     if (Err)
       return std::move(Err);
+
+    if (impl().NotifyCreated)
+      if (Error Err = impl().NotifyCreated(*J))
+        return Err;
```
Comment (joker-eph) [inline llvm/include/llvm/ExecutionEngine/Orc/LLJIT.h:493]: I don't know why the premerge CI didn't fail, but everywhere else in this function is `return std::move(Err);`

I made the change in 5d33f7176b00

Comment (joker-eph) [reply llvm/include/llvm/ExecutionEngine/Orc/LLJIT.h:493]: Seems like clang requires it but not gcc (which is used in premerge) nor MSVC.

You should have got a lot of buildbot emails though, didn't you?

Comment (weliveindetail) [reply llvm/include/llvm/ExecutionEngine/Orc/LLJIT.h:493]: I got a lot of buildbot emails, but they also came all day for all kinds of unrelated reasons... Anyway, sorry for the mess.

**Pull request merged**